cms/djangoapps/contentstore/views/course.py

```diff
@@ -1185,15 +1187,23 @@ def validate(self):
         if len(self.configuration.get('groups', [])) < 1:
             raise GroupConfigurationsValidationError(_("must have at least one group"))
 
-    def generate_id(self, used_ids):
+    @staticmethod
+    def generate_id(used_ids):
```

Comment:
I refactored the previous logic into common/djangoapps/util/db.py. Perhaps this could also live there, and then encapsulate the random_sql_int logic?